diff --git a/main.go b/main.go
index c5b138a..05a222e 100644
--- a/main.go
+++ b/main.go
@@ -57,7 +57,44 @@ type SMARTctlManagerCollector struct {
 
 // Describe sends the super-set of all possible descriptors of metrics
 func (i *SMARTctlManagerCollector) Describe(ch chan<- *prometheus.Desc) {
-	prometheus.DescribeByCollect(i, ch)
+	ch <- metricSmartctlVersion
+	ch <- metricDeviceModel
+	ch <- metricDeviceCount
+	ch <- metricDeviceCapacityBlocks
+	ch <- metricDeviceCapacityBytes
+	ch <- metricDeviceTotalCapacityBytes
+	ch <- metricDeviceBlockSize
+	ch <- metricDeviceInterfaceSpeed
+	ch <- metricDeviceAttribute
+	ch <- metricDevicePowerOnSeconds
+	ch <- metricDeviceRotationRate
+	ch <- metricDeviceTemperature
+	ch <- metricDevicePowerCycleCount
+	ch <- metricDevicePercentageUsed
+	ch <- metricDeviceAvailableSpare
+	ch <- metricDeviceAvailableSpareThreshold
+	ch <- metricDeviceCriticalWarning
+	ch <- metricDeviceMediaErrors
+	ch <- metricDeviceNumErrLogEntries
+	ch <- metricDeviceBytesRead
+	ch <- metricDeviceBytesWritten
+	ch <- metricDeviceSmartStatus
+	ch <- metricDeviceExitStatus
+	ch <- metricDeviceState
+	ch <- metricDeviceStatistics
+	ch <- metricDeviceErrorLogCount
+	ch <- metricDeviceSelfTestLogCount
+	ch <- metricDeviceSelfTestLogErrorCount
+	ch <- metricDeviceERCSeconds
+	ch <- metricSCSIGrownDefectList
+	ch <- metricReadErrorsCorrectedByRereadsRewrites
+	ch <- metricReadErrorsCorrectedByEccFast
+	ch <- metricReadErrorsCorrectedByEccDelayed
+	ch <- metricReadTotalUncorrectedErrors
+	ch <- metricWriteErrorsCorrectedByRereadsRewrites
+	ch <- metricWriteErrorsCorrectedByEccFast
+	ch <- metricWriteErrorsCorrectedByEccDelayed
+	ch <- metricWriteTotalUncorrectedErrors
 }
 
 // Collect is called by the Prometheus registry when collecting metrics.
